sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -103,12 +106,24 @@ public void process(TemplateInput templateInput, ProcessingEnvironment processin
         addImports(templateInput);
         addOrphanComments();
         setPackageDeclaration(packageName);
+        // Add class-level Javadoc instead of a line comment
+        compilationUnit.addOrphanComment(
+            new JavadocComment("This class is generated by annotation-processor. Initializes a new instance of the "
+                + serviceInterfaceImplShortName + " type."));
         createClass(serviceInterfaceImplShortName, serviceInterfaceShortName, templateInput);
 
         try (Writer fileWriter = processingEnv.getFiler()
             .createSourceFile(packageName + "." + serviceInterfaceImplShortName)
             .openWriter()) {
-            fileWriter.write(compilationUnit.toString());
+
+            // Convert the compilation unit into a string
+            String formattedCode = compilationUnit.toString();
+
+            // Post-process to add a blank line before `//` comments (starting with \n)
+            // The regex looks for a line break followed by `//` and adds a blank line before it.
+            formattedCode = formattedCode.replaceAll("(?<=\\n)(\\s*)//(?! Copyright|Licensed)(.*)", "\n$1//$2");
```

Comment:
There's a simpler fix for this issue which is just adding a leading space to the `LineComment` strings on lines 138 and 139

```java
        compilationUnit.addOrphanComment(new LineComment(" Copyright (c) Microsoft Corporation. All rights reserved."));
        compilationUnit.addOrphanComment(new LineComment(" Licensed under the MIT License."));
```

Reply:
yeah, that addresses the spacing issue. I was also trying to fix a feedback of introducing an empty line before LineComment but it was turning out lengthy so skipping doing that in this PR.